app/assets/javascripts/save_as.js

```diff
@@ -1,40 +1,77 @@
-/*globals document,customPrompt*/
+/*globals CustomPrompt*/
 
-$(document).on('page:change', function () {
+var SaveAsForm = (function () {
     'use strict';
 
-    var nameInput = $("input#testing_ground_name");
-
     function submitSaveAsForm() {
-        var form = $(this).parents("form");
-        form.attr("action", $(this).data('url'));
-        form.submit();
+        this.form
+            .attr("action", this.formAction)
+            .submit();
     }
 
-    function promptCallback(newName) {
-        nameInput.val(newName);
+    function isValid(value) {
+        return (value.length < 100 && value.length > 0);
```

Comment:
I removed this validation and instead use the server to validate the input as suggested.